shopify/resources/price_rule.py

```diff
@@ -0,0 +1,10 @@
+from ..base import ShopifyResource
+from .discount_code import DiscountCode
+
+class PriceRule(ShopifyResource):
+    def add_discount(self, discount_code = DiscountCode()):
```

Comment:
I'm not sure about this one either. A similar default param is used in draft_order.py in the send_draft_order_invoice method.  Is no default preferable in this case or is there a better way to do it?